REFERENCE.md

```diff
@@ -2495,6 +2526,48 @@ Possible strings for ExecStart, ExecStartPrep, ...
 
 Alias of `Variant[Enum[''], Pattern[/^[@\-:]*(\+|!|!!)?[@\-:]*\/.*/], Pattern[/^[@\-:]*(\+|!|!!)?[@\-:]*[^\/]*(\s.*)?$/]]`
 
+### <a name="Systemd--Unit--Slice"></a>`Systemd::Unit::Slice`
+
+Possible keys for the [Slice] section of a unit file
+
+* **See also**
+  * https://www.freedesktop.org/software/systemd/man/systemd.slice.html
+  * https://www.freedesktop.org/software/systemd/man/systemd.resource-control.html
+
+Alias of
+
+```puppet
+Struct[{
+    Optional['CPUAccounting']       => Boolean,
+    Optional['CPUQuota']            => Pattern['^([1-9][0-9]*)%$'],
+    Optional['CPUShares']           => Integer[2,262144],
+    Optional['CPUWeight']           => Variant[Enum['idle'],Integer[1,10000]],
+    Optional['Delegate']            => Boolean,
+    Optional['DeviceAllow']         => String[1],
+    Optional['DevicePolicy']        => Enum['auto','closed','strict'],
+    Optional['IOAccounting']        => Boolean,
+    Optional['IODeviceWeight']      => Array[Hash[Stdlib::Absolutepath, Integer[1,10000], 1, 1]],
+    Optional['IOReadBandwidthMax']  => Array[Hash[Stdlib::Absolutepath, Pattern['^(\d+(K|M|G|T)?)$'], 1, 1]],
+    Optional['IOReadIOPSMax']       => Array[Hash[Stdlib::Absolutepath, Pattern['^(\d+(K|M|G|T)?)$'], 1, 1]],
+    Optional['IOWeight']            => Integer[1,10000],
+    Optional['IOWriteBandwidthMax'] => Array[Hash[Stdlib::Absolutepath, Pattern['^(\d+(K|M|G|T)?)$'], 1, 1]],
+    Optional['IOWriteIOPSMax']      => Array[Hash[Stdlib::Absolutepath, Pattern['^(\d+(K|M|G|T)?)$'], 1, 1]],
+    Optional['IPAccounting']        => Boolean,
+    Optional['MemoryAccounting']    => Boolean,
+    Optional['MemoryHigh']          => Pattern['\A(infinity|\d+(K|M|G|T|%)?(:\d+(K|M|G|T|%)?)?)\z'],
+    Optional['MemoryLimit']         => Pattern['\A(infinity|\d+(K|M|G|T|%)?(:\d+(K|M|G|T|%)?)?)\z'], # dep'd in systemd
+    Optional['MemoryLow']           => Pattern['\A(infinity|\d+(K|M|G|T|%)?(:\d+(K|M|G|T|%)?)?)\z'],
+    Optional['MemoryMax']           => Pattern['\A(infinity|\d+(K|M|G|T|%)?(:\d+(K|M|G|T|%)?)?)\z'],
+    Optional['MemoryMin']           => Pattern['\A(infinity|\d+(K|M|G|T|%)?(:\d+(K|M|G|T|%)?)?)\z'],
+    Optional['MemorySwapMax']       => Pattern['\A(infinity|\d+(K|M|G|T|%)?(:\d+(K|M|G|T|%)?)?)\z'],
```

Comment:
DRY. Why not add a dedicated type for `Pattern['\A(infinity|\d+(K|M|G|T|%)?(:\d+(K|M|G|T|%)?)?)\z']`

Reply:
`user-.slice` acts as template for all user slices